darshan-util/pydarshan/darshan/report.py

```diff
@@ -660,15 +680,19 @@ def mod_read_all_apmpi_records(self, mod="APMPI", dtype=None, warnings=True):
             None
 
         """
+
+        if dtype is not None and _warnings:
+            warn(f"The use of the dtype argument is deprecated in v3.3.2 and will be removed in v3.4.x", DeprecationWarning)
```

Comment:
Why is the deprecation warning for `mod_read_all_apmpi_records` (and many others) gated on `_warnings`, but the deprecation warning for `mod_read_all_records` is not gated on `warnings`?